src/app/child-dev-project/schools/display-participants-count/display-participants-count.component.ts

```diff
@@ -0,0 +1,53 @@
+import { ChangeDetectorRef, Component, OnChanges } from "@angular/core";
+import { CommonModule } from "@angular/common";
+import { ChildrenService } from "../../children/children.service";
+import { ViewDirective } from "../../../core/entity/default-datatype/view.directive";
+import { DynamicComponent } from "../../../core/config/dynamic-components/dynamic-component.decorator";
+import { ChildSchoolRelation } from "../../children/model/childSchoolRelation";
+
+@DynamicComponent("DisplayParticipantsCount")
+@Component({
+  selector: "app-display-participants-count",
+  standalone: true,
+  imports: [CommonModule],
+  templateUrl: "./display-participants-count.component.html",
+})
+export class DisplayParticipantsCountComponent
+  extends ViewDirective<any>
+  implements OnChanges
+{
+  get participantRelationsCount(): number {
+    return this._participantRelationsCount;
+  }
+
+  set participantRelationsCount(value: number) {
+    this._participantRelationsCount = value;
+    this._changeDetectorRef.markForCheck();
+  }
+  private _participantRelationsCount: number;
+
+  constructor(
+    private _childrenService: ChildrenService,
+    private _changeDetectorRef: ChangeDetectorRef,
+  ) {
+    super();
+  }
+
+  override async ngOnChanges(): Promise<void> {
+    super.ngOnChanges();
+
+    return this._childrenService
+      .queryActiveRelationsOf("school", this.entity.getId())
+      .then(
+        (relations: ChildSchoolRelation[]) => {
+          this.participantRelationsCount = relations.length;
+        },
+        (reason) => {
+          console.log(
+            "Could not calculate participantRelationsCount, error response from ChildrenService.",
+            reason,
+          );
```

Comment:
Is there a reason why you use this promise syntax? So far we always you the distinct `then` and `catch` functions. I think this makes it much easier to understand what each function belongs to. In rxjs subscribers they also have deprecated the syntax for multiple functions as arguments.

Reply:
No, just a little rusty with JS ;)

Done.